errors.go

```diff
@@ -65,13 +65,35 @@ func NewInternalError(err error, msg string) *InternalError {
 	}
 }
 
+type InvalidArgumentErrorCode string
+
+const (
+	Accepted     InvalidArgumentErrorCode = "Accepted"     // Accepted Must be accepted
+	Blank        InvalidArgumentErrorCode = "Blank"        // Blank Can't be blank
+	Empty        InvalidArgumentErrorCode = "Empty"        // Empty Can't be empty
+	Even         InvalidArgumentErrorCode = "Even"         // Even Must be even
+	Exclusion    InvalidArgumentErrorCode = "Exclusion"    // Exclusion Is reserved
+	Inclusion    InvalidArgumentErrorCode = "Inclusion"    // Inclusion Is not included in the list
+	Invalid      InvalidArgumentErrorCode = "Invalid"      // Invalid Is invalid
+	NotANumber   InvalidArgumentErrorCode = "NotANumber"   // NotANumber Is not a number
+	NotAnInteger InvalidArgumentErrorCode = "NotAnInteger" // NotAnInteger Must be an integer
+	Odd          InvalidArgumentErrorCode = "Odd"          // Odd Must be odd
+	Present      InvalidArgumentErrorCode = "Present"      // Present Must be blank
+	Required     InvalidArgumentErrorCode = "Required"     // Required Must exist
+	Taken        InvalidArgumentErrorCode = "Taken"        // Taken Has already been taken
+)
+
+func (i InvalidArgumentErrorCode) String() string {
+	return string(i)
+}
+
 // InvalidArgumentError describes an invalid argument error.
 type InvalidArgumentError struct {
 	*BaseError
 
 	Kind       string
 	ID         string
-	Violations map[string][]string
+	Violations map[string][]InvalidArgumentErrorCode
```

Comment:
What do you think about accepting the `fmt.Stringer` interface here instead of our specific type? It will create a room for the developers to define their own set of errors (like we did here by implementing the `String()` on the enum) if the provided ones are not sufficient for them.